core/deployment/src/main/java/io/quarkus/deployment/console/AeshConsole.java

```diff
@@ -163,27 +182,27 @@ private void deadlockSafeWrite() {
     private void setup(Connection conn) {
         synchronized (this) {
             size = conn.size();
+            conn.setSignalHandler(event -> {
+                switch (event) {
+                    case INT:
+                        //todo: why does async exit not work here
+                        //Quarkus.asyncExit();
+                        //end(conn);
+                        new Thread(new Runnable() {
+                            @Override
+                            public void run() {
+                                System.exit(0);
+                            }
+                        }).start();
+                        break;
+                }
+            });
             if (inputSupport) {
-                conn.setSignalHandler(event -> {
-                    switch (event) {
-                        case INT:
-                            //todo: why does async exit not work here
-                            //Quarkus.asyncExit();
-                            //end(conn);
-                            new Thread(new Runnable() {
-                                @Override
-                                public void run() {
-                                    System.exit(0);
-                                }
-                            }).start();
-                            break;
-                    }
-                });
                 // Keyboard handling
                 conn.setStdinHandler(keys -> {
-                    InputHolder handler = inputHandlers.peek();
+                    var handler = inputHandler;
```

Comment:
First instance of `var` I've seen in the Quarkus codebase :)

Reply:
I only realized I could do it a few weeks ago, and I have put a few in :-)